connect/runtime/src/main/java/org/apache/kafka/connect/storage/ConnectorOffsetBackingStore.java

```diff
@@ -299,12 +349,11 @@ public Future<Void> set(Map<ByteBuffer, ByteBuffer> values, Callback<Void> callb
                     } catch (Exception e) {
                         log.warn("Failed to write offsets to secondary backing store", e);
                     }
+                    callback.onCompletion(null, null);
                 }
             }
-            try (LoggingContext context = loggingContext()) {
-                callback.onCompletion(primaryWriteError, ignored);
-            }
-        });
```

Comment:
Why is this part moved? It seems fine where it was, unless we're worried about dangling callbacks caused by potential exceptions being thrown above. But it also doesn't look like the new implementation is any better at preventing that from happening.

Reply:
Added it back.